src/main/java/com/vmware/vipclient/i18n/messages/service/ProductService.java

```diff
@@ -37,10 +36,9 @@ public ProductService(BaseDTO dto) {
      */
     public List<Map> getAllComponentTranslation() {
         List<Map> list = new ArrayList<Map>();
-        LocaleDTO localeDTO = new LocaleDTO();
-        List<String> locales = this.getSupportedLocales(VIPCfg.getInstance().getMsgOriginsQueue().iterator());
-        List<String> components = this.getComponents(VIPCfg.getInstance().getMsgOriginsQueue().iterator());
-        if (locales != null) {
+        List<String> locales = this.getSupportedLocales();
+        List<String> components = this.getComponents();
+        if (locales != null && components != null) {
             for (String languageTag : locales) {
                 for (Object component : components) {
                     MessagesDTO msgDTO = new MessagesDTO(((String) component).trim(), LocaleUtility.fmtToMappedLocale(Locale.forLanguageTag(languageTag)).toString().trim(),
```

Comment:
It's sure that component and locale are normalized. Don't need to do further procession.
```suggestion
                    MessagesDTO msgDTO = new MessagesDTO(component, languageTag, dto.getProductID(), dto.getVersion());
```

Reply:
Don't make this change as this is L3 code and not sure whether there will be error or not if don't do such process.